### ISSUE TITLE
ArrayIndexOutOfBoundsException when trying to get OCSP certificate from a loaded BDOC container

### ISSUE THREAD

**Issue by informatik01** [open-eid/digidoc4j#11]
HI.

I have found a bug.  When getting an OCSP certificate from a **BDOC** container that was previously loaded from a file or a stream an _ArrayIndexOutOfBoundsException_ exception is thrown in the _BDocSignature#getCN(X500Name x500Name)_ method. The reason is that inside that method there is no check for the size of the returned array of _org.bouncycastle.asn1.x500.RDN_ elements.:
```java
String name = x500Name.getRDNs(new ASN1ObjectIdentifier("2.5.4.3"))[0].getTypesAndValues()[0].getValue().toString();
```

This happens **ONLY if loading an existing BDOC container**. So if you create and sign a new BDOC container and then try to get an OCSP certificate, no exception is thrown. Also there is no such problem with **DDOC** containers.

---
Here is a simple code snippet to illustrate this issue:
```java
...
public static void main(String[] args) throws FileNotFoundException {
	Container container = ContainerBuilder.aContainer().fromExistingFile("example.bdoc").build();
	if (container != null) {
		if (container.getSignatures() != null && container.getSignatures().size() > 0) {
            System.out.println("Number of signatures: " + container.getSignatures().size());
				
			Signature sig = container.getSignatures().get(0);

			// Calling getOCSPCertificate() will cause ArrayIndexOutOfBoundsException
            // in BDocSignature#getCN(X500Name x500Name)
			X509Cert ocspCertificate = sig.getOCSPCertificate();
			...
		}
	}
}
```

**Comment by rvillido** [comment]
Hi, thanks for reporting it. I will investigate.

I created a pivotal ticket for this
https://www.pivotaltracker.com/story/show/111946527

**Comment by informatik01** [comment]
You're welcome!

**Comment by rvillido** [comment]
This is now fixed in the 1.0.2 RC release
https://github.com/open-eid/digidoc4j/releases/tag/public-1.0.2.rc
